internal/pkg/comm/creds.go

```diff
@@ -35,9 +35,9 @@ func NewServerTransportCredentials(
 	// NOTE: unlike the default grpc/credentials implementation, we do not
 	// clone the tls.Config which allows us to update it dynamically
 	serverConfig.config.NextProtos = alpnProtoStr
-	// override TLS version and ensure it is 1.2
+	// override TLS version and ensure it is 1.2 or 1.3
 	serverConfig.config.MinVersion = tls.VersionTLS12
-	serverConfig.config.MaxVersion = tls.VersionTLS12
+	serverConfig.config.MaxVersion = tls.VersionTLS13
```

Comment:
Why explicitly set the max version instead of relying on the judgement of the tls package's calculation of max?

Reply:
I just preserved what was here. Should I remove the Max, it will keep us from having to perform upkeep on it in the future as new TLS versions become available, but it also means we will blindly accept the consequences of any new version of TLS?

I prefer the explicit declaration of a max supported TLS version, but I'm indifferent on what we decide.